crates/sc-consensus-subspace-rpc/src/lib.rs

```diff
@@ -798,4 +810,30 @@ where
 
         Ok(last_segment_headers)
     }
+
+    // TODO:
+    // - the number of object mappings in each segment can be very large (hundreds or thousands).
+    //   To avoid RPC connection failures, limit the number of mappings returned in each response,
+    //   or the number of in-flight responses.
+    fn subscribe_archived_object_mappings(&self, pending: PendingSubscriptionSink) {
+        // There are no mappings in the genesis segment, so we don't need to recreate it here.
```

Comment:
I'm not sure this comment is helping much, there is nothing in this function that cares whether segment is genesis or any other at all :thinking:

Reply:
I've added a bit more context to the comment. Feel free to delete it totally if it's wrong or unnecessary, if you want to get this merged this week. (It's my Friday right now, so I might not see your next review until my Monday.)